### ISSUE TITLE
Installation of anvio 7 - Error building wheel

### ISSUE THREAD

**Issue by sarah872** [merenlab/anvio#1643]
## Short description of the problem

I cannot install anvio 7 due to an error when installing psutil, pysam & python-levenshtein through pip.

## System info

System Version: macOS 11.1 (20C69)
Kernel Version: Darwin 20.2.0

## Detailed description of the issue

I tried the following steps (as they are in the Installation help page):
```
conda env remove --name anvio-7
conda env create -f anvio-environment-7.yaml
conda activate anvio-7
conda install -y -c bioconda hmmer=3.2.1
curl -L https://github.com/merenlab/anvio/releases/download/v7/anvio-7.tar.gz --output anvio-7.tar.gz
pip install anvio-7.tar.gz
```
Here, the wheels for the packages `pysam`, `psutil`& `python-levenshtein` failed (please see attached log). However, I could install those through `conda`:
```
conda install -c conda-forge python-levenshtein  -y
conda install -c bioconda pysam  -y
conda install -c anaconda psutil==5.4.3 -y
pip install anvio-7.tar.gz
```
This finished successfully. After that, the program `anvi-self-test` wasn't found, but this helped:
```
conda deactivate
conda activate anvio-7
```

However, the version is wrong, although it's at the right location, and it can't be run because a module is missing:
```
anvi-self-test -v                                                                                                                                 
Anvi'o version ...............................: esther (v6.2-master)
Profile DB version ...........................: 34
Contigs DB version ...........................: 18
Pan DB version ...............................: 14
Genome data storage version ..................: 7
Auxiliary data storage version ...............: 2
Structure DB version .........................: 2
Kegg Modules DB version ......................: 2

which anvi-self-test                                                                                                                              
/Users/user/miniconda3/envs/anvio-7/bin/anvi-self-test

anvi-self-test                                                                                                 
Traceback (most recent call last):
  File "/Users/user/miniconda3/envs/anvio-7/bin/anvi-self-test", line 91, in <module>
    from anvio.argparse import ArgumentParser
ModuleNotFoundError: No module named 'anvio.argparse'
```
[anvio-install.log](https://github.com/merenlab/anvio/files/5820538/anvio-install.log)

**Comment by mschecht** [comment]
Hi @sarah872, I'm sorry you are having issues with your installation. Thank you very much for attaching the complete log file :)

Your conda env is linked to an older c-compiler on your mac: `x86_64-apple-darwin13.4.0-clang`. I would recommend linking your conda env to the newest version available for mac called `x86_64-apple-darwin20.2.0`. I addressed a similar error in a separate issue #1636. 

Please try to emulate the solution in #1636 and if you are still having issues please let us know.

**Comment by sarah872** [comment]
Thanks for pointing out the old `clang` version.
The `sudo rm -rf /Library/Developer/CommandLineTools && xcode-select --install` unfortunately did not help, as pip still used the old `clang` to compile. However, I managed after 
1. deleting all other anvio-related conda environments (anvio-6.2, anvio-6.3, anvio-master)
2. removing the github copy of `anvio-master` (mine was in `~/github/anvio`)
3. deleting pip cache (`rm -rf ~/Library/Caches/pip/*`)
4. Following the steps as above, with a final `env CC=/usr/bin/CC pip install anvio-7.tar.gz`.

Thanks!
